<!-- -*- mode: XML -*- -->
<!-- run wanji_pointcloud/CloudNodelet in a nodelet manager -->

<launch>
  <arg name="calibration" default="" />
  <arg name="manager" default="wanji_nodelet_manager" />
  <arg name="max_angle" default="360.0" />
  <arg name="min_angle" default="0" />
  <arg name="max_distance" default="70.0" />
  <arg name="min_distance" default="0.5" />
  <arg name="rpm" default="600" />
  <arg name="return_mode" default="1" />
  <arg name="time_mode" default="true" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)_cloud"
        args="load wanji_pointcloud/CloudNodelet $(arg manager)">
    <param name="calibration" value="$(arg calibration)"/>
    <param name="max_angle" value="$(arg max_angle)"/>
    <param name="min_angle" value="$(arg min_angle)"/>
    <param name="max_distance" value="$(arg max_distance)"/>
    <param name="min_distance" value="$(arg min_distance)"/>
    <param name="rpm" value="$(arg rpm)"/>
    <param name="return_mode" value="$(arg return_mode)"/>
    <param name="time_mode" value="$(arg time_mode)"/>
  </node>
</launch>
